source/frontend/src/features/leases/list/LeaseSearchResults/LeaseSearchResults.tsx

```diff
@@ -109,6 +111,94 @@ const columns: ColumnWithProps<ApiGen_Concepts_Lease>[] = [
       );
     },
   },
+  {
+    Header: 'Historical File #',
+    align: 'left',
+    clickable: false,
+    sortable: false,
+    width: 10,
+    maxWidth: 20,
+    Cell: (props: CellProps<ApiGen_Concepts_Lease>) => {
+      // File numbers types to display
+      const numberTypes: string[] = [
+        ApiGen_CodeTypes_HistoricalFileNumberTypes.LISNO.toString(),
+        ApiGen_CodeTypes_HistoricalFileNumberTypes.PSNO.toString(),
+        ApiGen_CodeTypes_HistoricalFileNumberTypes.OTHER.toString(),
+      ];
+
+      // Get unique file numbers from lease properties
+      const fileNumbers: ApiGen_Concepts_HistoricalFileNumber[] = [];
+      props.row.original.fileProperties?.forEach(fl => {
+        fl.property.fileNumbers?.forEach(number => {
+          if (numberTypes.includes(number.fileNumberTypeCode.id)) {
+            if (
+              !fileNumbers.find(
+                x =>
+                  x.fileNumber === number.fileNumber &&
+                  x.fileNumberTypeCode.id === number.fileNumberTypeCode.id,
+              )
+            ) {
+              fileNumbers.push(number);
+            }
+          }
+        });
+      });
+
+      // group numbers by type
+      const numbersByType = fileNumbers.reduce((acc, number) => {
```

Comment:
nit: could replace this with a group by, as that appears to be what you want here.

Reply:
tried it, but I ran with some issues during test execution.